pyserini/encode/_models.py

```diff
@@ -0,0 +1,122 @@
+#
```

Comment:
What's the difference between `_models` and `_doc_encoders`?

Reply:
_models is to define custom hgf models.

models are used by encoders.

but I'll group it into encoders as we discussed vvvv